doc/src/manual/faq.md

```diff
@@ -556,6 +556,26 @@ julia> remotecall_fetch(anon_bar, 2)
 1
 ```
 
+## Troubleshooting "method not matched" (parametric invariance, etc.)
+
+### Why doesn't it work to declare `foo(bar::Vector{Real}) = 42` and then call `foo([1])`?
+
+As you'll see if you try this, the result is
+
+```julia
+MethodError: no method matching foo(::Array{Float64,1})
+Closest candidates are:
+  foo(!Matched::Array{real,1})
```

Comment:
`Real` upper case, and if you call it on `[1]` then the no-method error will say `foo(::Array{Int64,1})` (or `Int32` on 32 bit)

this could fit well as a doctest example